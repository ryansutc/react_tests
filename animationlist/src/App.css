.App {
  text-align: left;
  margin: 20px;
}

td {
  padding-right: 48px;
  padding-left: 12px;
  border: solid;
}
tr {
  width: 220px;
}
.App-header {
  display: absolute;
  left: 50%;
  top: 50%;
  font-size: 16px;
  color: white;
}

.App-link {
  color: #61dafb;
}

body {
  background-color: #282c34;
}
