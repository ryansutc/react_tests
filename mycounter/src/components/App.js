import React from 'react'

import Counter from './Counter'
import Posts from './Posts'
import Header from '../Header';
import { connect } from 'react-redux';
import { container } from '@material-ui/core'
const App = (store) => {
  return (
    <div>
      <Counter />
      <Posts />
    </div>
  )
}

export default connect()(App);

