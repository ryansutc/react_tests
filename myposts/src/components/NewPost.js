import React from 'react';

class NewPost extends React.Component {
    state = {
        title: '',
        body: ''
    };

    // our onChange event handler. Needed for React:
    // further reading here: https://medium.com/capital-one-tech/how-to-work-with-forms-inputs-and-events-in-react-c337171b923b
    handleInputChange = e => {
        this.setState({
            [e.target.name]: e.target.value
        }); //Note to self: setState allows component to control its own state, but does not involve redux in any way.
    };

    handleSubmit = (e) => {
        e.preventDefault();
        
        if (this.state.title.trim() && this.state.body.trim()) {
            console.log(this.state);
            this.props.onAddPost({title: this.state.title, body: this.state.body}); // here we call the onAddPost property (a function) which will send the new state info to Redux state container via an action.
            this.handleReset();
        }
    }

    handleReset = () => {
        this.setState({
            title: '',
            body: ''
        });
    };

    render() {
        return (
            <div data-testid="form1">
                <form onSubmit={this.handleSubmit }>
                    <div className="form-group">
                        <input 
                            type="text" 
                            placeholder="title" 
                            className="form-control" 
                            name="title"
                            onChange={ this.handleInputChange }
                            value={ this.state.title }
                        />
                    </div>
                    <div className="form-group">
                        <textarea
                            cols="15" rows="8" placeholder="Body" className="form-control" 
                            name="body"
                            onChange={ this.handleInputChange }
                            value={ this.state.body} >
                        </textarea>
                    </div>
                    <div className="form-group">
                        <button type="submit" className="btn btn-primary">Add Post</button>
                        <button type="button" className="btn btn-warning" onClick={ this.handleReset }>Reset</button>
                    </div>
                </form>
          </div>
        )
    }
}

export default NewPost;