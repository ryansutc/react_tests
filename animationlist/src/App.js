import "./App.css";

import React, { useState } from "react";

import { animated, useTransition } from "@react-spring/web";

import { LISTS } from "./data.js";
import { usePrevious } from "./utils.js";

/**
 * The main react parent component
 *
 * @returns {Object} React.Component
 */
function App() {
  const [curList, setCurList] = useState(0);
  const prevList = usePrevious(curList);

  const listTransition = useTransition(LISTS[curList], {
    // enter: (item) => (next) =>
    //   next({ delay: 700, marginTop: 5, maxHeight: 50, opacity: 1 }),
    //expires: 400,
    enter: { delay: 800, marginTop: 5, maxHeight: 50, opacity: 1 },
    from: { marginTop: 5, opacity: 0 },
    keys: (item) => item.id,
    leave: { delay: 0, marginTop: 0, maxHeight: 0, opacity: 0 },
    onRest: {
      opacity: (props, spring, item) => {
        console.log("At Rest: Give newListProps. " + JSON.stringify(props));
      },
    },
    loop: () => {
      console.log("I have done a run");
      return false;
    },
    // onStart: () =>
    //   console.log("At Start: Give newListNames, deltaNewListProps"),
    //trail: 300,
    expires: 10,
    //onProps: (props, spring) =>
    //console.log("I've been updated " + JSON.stringify(spring)),
  });

  const animatedList = listTransition((style, item) => {
    return (
      item && (
        <animated.tr style={style} data={item.id}>
          <td style={{ width: "80px" }}>{item.name}</td>
          <td style={{ width: "140px" }}>{item.status}</td>
        </animated.tr>
      )
    );
  });

  const lastAnimatedItem = () => {
    let oldList = prevList ? LISTS[prevList] : null;
  };
  let height =
    Math.max(
      prevList ? LISTS[prevList].length : LISTS[0].length,
      LISTS[curList].length
    ) * 50;

  //  style={{ height: height }}
  console.log("rendering");
  return (
    <div className="App">
      <table className="App-header">
        <tbody>{animatedList}</tbody>
      </table>
      <p>
        <button onClick={() => setCurList(curList === 3 ? 0 : curList + 1)}>
          Click Me
        </button>
      </p>
      <p>{curList}</p>
      {/* <SwitchThing /> */}
    </div>
  );
}

export default App;
