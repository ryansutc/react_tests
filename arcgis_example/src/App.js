import React from 'react';
import './App.css';
import { Map } from '@esri/react-arcgis';

import FeatureLayer from './FeatureLayer';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      map: null,
      view: null,
      featureLayers: []
    };

    this.handleMapLoad = this.handleMapLoad.bind(this);
    this.handleFail = this.handleFail.bind(this);
    this.createFeatureLayers = this.createFeatureLayers.bind(this);
    this.featureLayers = ["https://services8.arcgis.com/vVBb77z9fDbXITgG/ArcGIS/rest/services/SamplePoints/FeatureServer/0"];
  }

  render() {
    console.log("map is rendering: " + this.state)
    return (
      <div style={{ width: '100vw', height: '100vh' }}>
        <Map
          onLoad={this.handleMapLoad}
          onFail={this.handleFail}
          mapProperties={{
            basemap: 'dark-gray',
          }}
          viewProperties={{
            center: [-123.1257, 49.249],
            zoom: 17
          }}>
            { this.createFeatureLayers() }
        </Map>
      </div>
    );
  }

  createFeatureLayers(){
    let featureLayerComponents = []
    for (var layer in this.state.featureLayers) {
      featureLayerComponents.push(<FeatureLayer layerID={ this.state.featureLayers[layer] }/>)
    }
    return featureLayerComponents
  }


  handleMapLoad(map, view) {
    console.log('Map Loaded.')
    this.setState({ map: map, view: view, featureLayers: this.featureLayers });
  }

  handleFail(e) {
    console.error(e);
    this.setState({ status: 'failed' });
  }
}

export default App;
