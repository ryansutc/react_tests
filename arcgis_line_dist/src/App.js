import React from 'react';
import ReactDOM from 'react-dom';
import { loadModules } from 'esri-loader';
import './App.css';

import SketchGraphicContainer from './SketchGraphicContainer';
import CoordinateWidget from './CoordinateWidget';
import StatusBox from './StatusBox';
import QueryButton from './QueryButton';
import { Map } from '@esri/react-arcgis';

import FeatureLayer from './FeatureLayer';
import { getSymbolForPt, updateSelectedPts, addSamplePts } from './MapUtils';
import { getLengthOfLine, getSampleCoordsForPolyline } from './GeomUtils';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      dist: 20,
      map: null,
      view: null,
      featureLayers: [],
      selectedPts: null,
      sketchState: "not started",
      sketchLength: 0,
      samplePtsGeom: null,
      layerLoaded: false,
      samplePtsGraphicsLayer: null, 
      transectLinesGraphicsLayer: null
    };

    this.handleMapLoad = this.handleMapLoad.bind(this);
    this.handleMapClick = this.handleMapClick.bind(this);
    this.handleFail = this.handleFail.bind(this);
    this.createFeatureLayers = this.createFeatureLayers.bind(this);
    this.renderViewContent = this.renderViewContent.bind(this);
    this.featureLayers = ["https://services8.arcgis.com/vVBb77z9fDbXITgG/ArcGIS/rest/services/SamplePoints/FeatureServer/0"];
  }

  render() {

    return (
      <div style={{ width: '100vw', height: '100vh' }}>
        <Map
          onLoad={this.handleMapLoad}
          onFail={this.handleFail}
          mapProperties={{
            basemap: 'dark-gray',
          }}
          loadElement={"loading spacecraft..."}
          viewProperties={{
            center: [-123.1257, 49.249],
            zoom: 4
          }}>
          {this.createFeatureLayers()}

          <StatusBox
            status={this.state.sketchState}
            distance={this.state.sketchLength}
          />
          {this.state.layerLoaded && this.state.samplePtsGraphicsLayer ?
            <QueryButton
              loading={false} //if we've got featureLayers, we're loaded?
              featureLayer={this.state.featureLayers[0]}
              samplePtsGraphicsLayer={this.state.samplePtsGraphicsLayer}
              dist={this.state.dist}
            /> : <div />
          }
          
        </Map>
      </div>
    );
  }
  /**
   * This method fires once the map view is loaded & ready.
   * It adds widgets and items to view
   * @param {*} view 
   */
  renderViewContent(view) {
    const node_br = document.createElement("div");
    view.ui.add(node_br, "bottom-right");

    ReactDOM.render(
      <SketchGraphicContainer
        map={this.state.map}
        view={this.state.view}
        selectPts={this.selectPts}
        handleSelectPts={this.handleSelectPts}
        create={this.handleSketchCreate.bind(this)}
        measure={this.handleMeasure.bind(this)}
        samplePtsGeom={this.state.samplePtsGeom}
        transectLinesGraphicsLayer={this.state.transectLinesGraphicsLayer}
      />, node_br
    );

    CoordinateWidget({
      map: this.state.map,
      view: this.state.view
    }).then((ccWidget) => {
      view.ui.add(ccWidget, "bottom-left")
    });
  }

  handleMapClick(event) {
    this.state.view.hitTest(event).then(response => {
      if (response.results.length) {
        let features = response.results.filter(result => {
          return result.graphic.layer.layerId === 0
        })
        features.forEach(feature => feature.graphic.symbol = getSymbolForPt(feature));
        this.handleSelectPts([features[0].graphic], event.native.shiftKey);
      }
    });
  };

  handleMeasure(event) {
    console.log("handle measure is firing!");
    console.log(getLengthOfLine(event.graphic.geometry.paths[0]));
  }

  handleSketchCreate(event) {
    if (event.state === "complete") {
      if (this.state.sketchState !== "complete") {
        this.setState({ sketchState: "complete" });
        let polylineGeom = event.graphic.geometry.paths[0];
        let samplePtGeoms = getSampleCoordsForPolyline(polylineGeom, this.state.dist);

        this.setState({ samplePtsGeom: samplePtGeoms });
        //add the samplePtGeom coords to map view as graphics
        addSamplePts(samplePtGeoms, this.state.samplePtsGraphicsLayer, this.state.dist);
      }
    }
    if (event.state === "active") {
      if (this.state.sketchState !== "active") {
        this.setState({ sketchState: "active" });
      }
      if (event.toolEventInfo.type === "cursor-update") {
        this.setState({ sketchLength: getLengthOfLine(event.graphic.geometry.paths[0])})
      }
    }

    if (event.state === "cancel") {
      this.setState({ sketchState: "cancel" });
    }

    if (event.state === "start") {
      if (this.state.sketchState !== "start") {
        this.setState({ sketchState: "start" });
        this.state.samplePtsGraphicsLayer.removeAll();
        this.state.transectLinesGraphicsLayer.removeAll();
      }
    }
  }

  handleSelectPts(newSelection, shift = false) {

    updateSelectedPts(this.state.selectedPts, newSelection, shift).then((updatedSelectedPts) => {
      this.state.view.graphics.removeAll();
      this.state.view.graphics.addMany(updatedSelectedPts)
      this.setState({ selectedPts: updatedSelectedPts })
    });
  }

  createFeatureLayers() {
    let featureLayerComponents = []
    console.log("create Feature layers called");
    for (var layer in this.featureLayers) {
      featureLayerComponents.push(
        <FeatureLayer key={"1"}
          layerID={this.featureLayers[layer]}
          layerLoaded={this.handleLayersLoaded.bind(this)}
        />)
    }
    return featureLayerComponents
  }

  handleLayersLoaded(featureLayer) {
    console.log("feature layer loaded " + featureLayer);
    this.setState({layerLoaded: true, featureLayers: [featureLayer]});
  }

  handleMapLoad(map, view) {
    loadModules([
      'esri/layers/GraphicsLayer'
    ])
    .then(([GraphicsLayer]) => {
      var samplePtsGraphicsLayer = new GraphicsLayer({
        graphics: [],
        title: "Sample Pt Centroids",
        id: "SamplePts"
      });

      var transectLinesGraphicsLayer = new GraphicsLayer({
        graphics: [],
        title: "Transect Lines",
        id: "TransectLines"
      });

      map.layers.addMany([samplePtsGraphicsLayer, transectLinesGraphicsLayer]);

      console.log('Map Loaded.')
      this.setState({ map: map, view: view, featureLayers: this.featureLayers, 
        samplePtsGraphicsLayer: samplePtsGraphicsLayer,
        transectLinesGraphicsLayer: transectLinesGraphicsLayer
      });
      this.handleViewLoad(view);
    });
  }

  handleViewLoad(view) {
    view.when(() => {
      view.on("click", this.handleMapClick);
      this.renderViewContent(view);
    })
  }

  handleFail(e) {
    console.error(e);
    this.setState({ status: 'failed' });
  }
}



export default App;
