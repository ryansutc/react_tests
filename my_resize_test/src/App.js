import React from 'react';
import './App.css';
import Hello from "./Hello";


class App extends React.Component {
  render() {
    return (
    <div className="App">
      <Hello />
    </div>
    )
  }
}

export default App;
