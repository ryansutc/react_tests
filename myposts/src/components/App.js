import React from 'react';

import CreatePost from '../containers/CreatePost';
import './App.css';
import '../../node_modules/bootstrap/dist/css/bootstrap.min.css';

import PostList from '../containers/PostList';

function App() {
  return (
    <div className="container">
      <div className="row">
        <div className="col-md-6">
          <CreatePost />
        </div>
        <div className="col-md-4">
          <PostList />
        </div>
      </div>
    </div>
  );
}

export default App;
