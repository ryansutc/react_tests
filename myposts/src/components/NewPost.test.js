import '@testing-library/react/cleanup-after-each';
import '@testing-library/jest-dom/extend-expect';
import {render, fireEvent, Queries} from '@testing-library/react';

import React from 'react';
import NewPost from './NewPost';
import CreatePost from '../containers/CreatePost';
import { Provider } from 'react-redux';
import ReactDOM from 'react-dom';
import App from './App';

import { createStore } from 'redux';
import rootReducer from '../reducers';


// normally here you'd do:
// const div = document.createElement('div');
//     const store = createStore(
//         rootReducer,
//         window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
//         );
        
//         ReactDOM.render(
//             <Provider store={store}>
//                 <CreatePost />
//             </Provider>, div);
// but for this test we'll umm... not do that :)

// Now here's what your test will look like:

function renderWithRedux(ui,
  {initialState, store = createStore(rootReducer, initialState)} = {},
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store,
  }
}

test('show that a new post was created on create Post click', () => {
    const {getByLabelText, getByPlaceholderText, getByText, container} = renderWithRedux(
        <App />,
    )
        //okay, we want to add title and body to the NewPost form, can 
        // we just update state? Or we gotta mock filling form?
        
        let message = 'hey there, this is the body';
        //see: https://testing-library.com/docs/guide-which-query
        fireEvent.change(getByPlaceholderText(/title/i), {target: {value: 'chuck norris'}})
        fireEvent.change(getByPlaceholderText(/body/i), {target: {value: message}})
            
        fireEvent.click(getByText("Add Post"))
        
        //ReactDOM.unmountComponentAtNode(div);
        expect(getByText(message)).toBeInTheDocument();
})