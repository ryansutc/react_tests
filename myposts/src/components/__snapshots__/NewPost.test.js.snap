// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly. Snapshot test 1`] = `
<div
  className="Post"
  id="post5"
>
  <h2>
    hey you
  </h2>
  <div />
  <button
    className="btn btn-danger"
    onClick={[Function]}
    type="button"
  >
    Remove
  </button>
</div>
`;
